### ISSUE TITLE
Statements are not aligned for ASI style

### ISSUE THREAD

**Issue by unional** [palantir/tslint#1490]
### Bug Report
- **TSLint version**: `3.14.0`
- **TypeScript version**: `2.0.0`
- **Running TSLint via**: (please choose one) CLI
#### TypeScript code being linted

https://github.com/unional/jspm-config/blob/master/src/readJspmConfig.ts#L15-L20
https://travis-ci.org/unional/jspm-config/builds/152584793

Simplified example:

``` ts
function foo() {
  callSomething()

  ;(process as any).doSomething()
}
```

Expect this would pass lint. 🌷

**Comment by unional** [comment]
Likely from TS?
https://github.com/Microsoft/TypeScript/issues/10392

**Comment by JKillian** [comment]
So the real issue here, perhaps, is that the `semicolon` rule, when using option `never`, doesn't allow this?

```
function foo() {
  callSomething();
  (process as any).doSomething()
}
```

**Comment by unional** [comment]
That would be nice, if it cannot be solve on the TS side, this is the least intrusive solution. 👍

**Comment by JoshuaKGoldberg** [comment]
Sounds like the `semicolon` rule should have an option to allow semicolons when it would interact with automatic semicolon insertion (ASI).

**Comment by JoshuaKGoldberg** [comment]
☠️ *TSLint's time has come!* ☠️ 

TSLint is no longer accepting most feature requests per #4534. See **[typescript-eslint.io](https://typescript-eslint.io)** for the new, shiny way to lint your TypeScript code with **ESLint**. ✨ 

It was a pleasure open sourcing with you all!

**Comment by JoshuaKGoldberg** [comment]
🤖 Beep boop! 👉  **TSLint is deprecated** 👈 _(#4534)_  and you should **switch to [typescript-eslint](https://typescript-eslint.io)**! 🤖 

🔒 This issue is being locked to prevent further unnecessary discussions. Thank you! 👋
